File: DotNetKamp/GameStoreApplication/GameStoreApplication/Concrete/CampaignManager.cs
using System;
using System.Collections.Generic;
using System.Text;
using GameStoreApplication.Abstract;
using GameStoreApplication.Entities;

namespace GameStoreApplication.Concrete
{
    public class CampaignManager:ICampaignService
    {
        private List<Campaign> _campaigns = new List<Campaign>();
        public void Add(Campaign campaign)
        {
            _campaigns.Add(campaign);
        }

        public void Update(Campaign campaign)
        {
            foreach (var cmpgn in _campaigns)
            {
                if (cmpgn == campaign)
                {
                    int index = _campaigns.IndexOf(cmpgn);
                    Console.WriteLine("{0} indexteki {1} id {2} adında ki kampanyanın indirim oranı: {3}", index,
                        cmpgn.Id, cmpgn.CampaignName, cmpgn.DiscountPercentage);
                    Console.Write("Neyi güncellemek istiyorsunuz? 1- Kampanya Adı 2-İndirim Oranı");
                    var secim = Console.ReadLine();
                    int secimInt = Convert.ToInt32(secim);
                    if (secimInt == 1)
                    {
                        Console.Write("Yeni adı giriniz = ");
                        var updateGameName = Console.ReadLine();
                        _campaigns[index].CampaignName = updateGameName;

                    }
                    else if (secimInt == 2)
                    {
                        Console.Write("Yeni Stok giriniz = ");
                        var updateStock = Console.ReadLine();
                        _campaigns[index].DiscountPercentage = Convert.ToInt32(updateStock);

                    }
                    else
                    {
                        Console.WriteLine("Hatalı seçim yaptınız!");
                    }
                    Get(cmpgn);
                    break;
                }

            }
            Console.WriteLine("Kayıt Bulunamadı!!!!");
        }

        public void Delete(Campaign campaign)
        {
            foreach (var cmpgn in _campaigns)
            {
                if (cmpgn == campaign)
                {
                    int index = _campaigns.IndexOf(cmpgn);
                    Console.WriteLine("{0} indexteki {1} id kampanya silindi!", index, cmpgn.Id);
                    _campaigns.Remove(cmpgn);
                    break;
                }
                else
                {
                    Console.WriteLine("Kayıt Bulunamadı!!!!");
                }
            }
            
        }

        public void GetAll()
        {
            Console.WriteLine("----------- Tüm Kampanyalar -------------");
            foreach (Campaign cmpgn in _campaigns)
            {
                Console.WriteLine("{0} id, adı {1}, İndirim oranı: {2}",
                    cmpgn.Id, cmpgn.CampaignName, cmpgn.DiscountPercentage);
            }
            Console.WriteLine("----------- --------------- -------------");
        }

        public void Get(Campaign campaign)
        {
            throw new NotImplementedException();
        }
    }
}

File: JavaKamp/JavaOdevler/4.GunOdev3/Concretes/CampaignManager.java
package Concrete;

import java.util.ArrayList;
import java.util.List;

import Abstract.CampaignService;
import Entities.Campaign;

public class CampaignManager implements CampaignService {
	List<Campaign> campaigns = new ArrayList<Campaign>();
	

	@Override
	public void add(Campaign campaign) {
		this.campaigns.add(campaign);
		System.out.println(campaign.getName() + " kampanya eklendi!");
	}

	@Override
	public void update(Campaign campaign) {
		
			boolean dataGeldiMi=false;
			int index = 0;
			for (int i=0; i<this.campaigns.size();i++) {
				Campaign campaign1 = this.campaigns.get(i);
				if (campaign.getId()==campaign1.getId()) {
					index = i;
					dataGeldiMi=true;
					break;
				}
			}
			
			if (dataGeldiMi) {
				this.campaigns.set(index, campaign);
				System.out.println(campaign.getName() + " kampanya güncellendi!");
			}else {			
				System.err.println("Data bulunamadý!");
			}
		

	}

	@Override
	public void delete(Campaign campaign) {
		this.campaigns.remove(campaign);
		System.out.println(campaign.getName() + " kampanya silindi!");
	}

	@Override
	public void getAll() {
		for (Campaign campaign : campaigns) {
			System.out.println("Id : " + campaign.getId() + " Adý : " + campaign.getName() + " Ýndirim kodu : "
					+ campaign.getDiscountCode() + " Ýndirim Oraný% : " + campaign.getDiscountRate());
		}

	}

	@Override
	public Campaign getByDiscountCode(String discountCode) {
		int index = 0;
		boolean dataGeldiMi=false;
		for (int i=0; i<this.campaigns.size();i++) {
			Campaign campaign1 = this.campaigns.get(i);
			if (discountCode==campaign1.getDiscountCode()) {
				index = i;
				dataGeldiMi =true;
				break;
			}
		}
		if (dataGeldiMi) {
			return this.campaigns.get(index);
		}else {		
			System.err.println("Data bulunamadý!");
			return null;
		}
		
	}

}
